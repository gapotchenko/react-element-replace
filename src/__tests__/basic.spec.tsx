import * as React from 'react';
import { Replacer } from "..";
import { create } from 'react-test-renderer';

test("number replace", () => {
    const component = create(
        <Replacer 
            match={x => typeof x === 'number'} 
            replace={(x: number) => <>{x + 1}</>}>
                {1}{2}{3}
        </Replacer>
    );
    let tree = component.toJSON();
    expect(tree).toStrictEqual(['2','3','4']);
});

test("basic div replace", () => {
    const component = create(
        <Replacer
            match={(x: {type: any}) => x.type === 'div'}
            replace={() => <>{"This was a div"}</>}>
            <div>Replace Me</div>
            <span>Don't Replace Me</span>
        </Replacer>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("no-op replace", () => {
    const component = create(
        <Replacer
            match={() => false}
            replace={() => <></>}>
            <div>
                <div>1</div>
                <div>
                    <div>2</div>
                </div>
            </div>
        </Replacer>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

