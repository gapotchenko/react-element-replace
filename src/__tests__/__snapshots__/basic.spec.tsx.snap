// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic div replace 1`] = `
Array [
  "This was a div",
  <span>
    Don't Replace Me
  </span>,
]
`;

exports[`no-op replace 1`] = `
<div>
  <div>
    1
  </div>
  <div>
    <div>
      2
    </div>
  </div>
</div>
`;
